### ISSUE TITLE
Purge API - how to delete messages and media content?

### ISSUE THREAD

**Issue by sorcer1122** [matrix-org/synapse#10228]
Hi guys,

I am struggling with API. Want to make a cron task that will clean up a specific room (delete ALL messages) say every 24h. I am using the following command:

`curl --header "Authorization: Bearer [my token]" -XPOST -H "Content-Type: application/json" -d '{ "purge_up_to_ts": 36000 }' "http://localhost:8003/_synapse/admin/v1/purge_history/[room_id]"`

I get the following response - {"errcode":"M_NOT_FOUND","error":"there is no event to be purged"}

Similar issue happens with media folder. I use the following command:

`curl --header "Authorization: Bearer [my token]" -XPOST "http://localhost:8003/_synapse/admin/v1/media/[server_name]/delete?before_ts=36000"` and I get "{"deleted_media":[],"total":0}"

Any advice on how to fix it would be highly appreciated.

**Comment by dklimpel** [comment]
Please check your timestamps.
You need something like this: `curl -H "Authorization: Bearer <access_token>" -d '' \
  "http://localhost:8008/_synapse/admin/v1/purge_media_cache?before_ts=$(date -d-6months +%s)000"`

**Comment by dklimpel** [comment]
You want to delete everything that is older than 1h and use the timestamp `36000`.
With this timestamp you say to Synapse delete everything older than `Thu Jan 01 1970 00:00:36`.
The timestamp you need for the API is UNIX time in milliseconds.
Delete everything older than `22 Jun 2021` you need this timestamp: `1624320000000`

**Comment by sorcer1122** [comment]
@dklimpel 
Yes, I understand the timestamp issue but even using 1624320000000 for timestamp gives me 'curl no URL specified!' error.

**Comment by dklimpel** [comment]
> @dklimpel
> Yes, I understand the timestamp issue but even using 1624320000000 for timestamp gives me 'curl no URL specified!' error.

This is a curl problem. But without full command line there is no chance for help.
Also the is a help romm in matrix: **#synapse:matrix.org**
